tests/scalability/generate_brac_data/generate-brac-data.js

```diff
@@ -0,0 +1,217 @@
+const preconditionDataDirectory = dataConfig.data_directory
+  + dataConfig.precondition_data_directory + dataConfig.json_directory;
+const mainDataDirectory = dataConfig.data_directory + dataConfig.main_script_data_directory;
+const usersDirectory = dataConfig.data_directory + dataConfig.precondition_data_directory;
+const dataExtension = dataConfig.json_data_extension;
+const csvWriter = createCsvWriter({
+  path: usersDirectory + 'users.csv',
+  header: [
+    { id: 'username', title: 'username' },
+    { id: 'password', title: 'password' },
+    { id: 'roles', title: 'roles' },
+    { id: 'contact', title: 'contact' },
+    { id: 'phone', title: 'phone' },
+    { id: 'place', title: 'place' },
+  ],
+  alwaysQuote: true,
+  append: true
+});
+
+const users = [];
+const managers = [];
+
+const createDataDoc = (directory, content) => {
+  fs.writeFile(directory, content, err => {
+    if (err) {
+      console.error('CreateDataDoc ' + err);
+      return;
+    }
+  });
+};
+
+const createDataDirectory = (path, directoryName) => {
+  if (!fs.existsSync(path + directoryName)) {
+    fs.mkdirSync(path + directoryName, err => {
+      if (err) {
+        console.error('CreateDataDirectory ' + directoryName + ' - ' + err);
+        return;
+      }
+    });
+  }
+};
+
+const setParents = (firstLevelId, secondLevelId, thirdLevelId) => {
+  return {
+    _id: firstLevelId,
+    parent: {
+      _id: secondLevelId,
+      parent: {
+        _id: thirdLevelId
+      }
+    }
+  };
+};
+
+const pairPlaceTypesRoles = {
+  'district_hospital': 'supervisor',
+  'health_center': 'district_admin',
+  'clinic': null
+};
+
+const pairPlaceTypesNeedsUsers = {
+  'district_hospital': true,
+  'health_center': true,
+  'clinic': false
+};
+
+const pairPlaceTypesPersonSubtype = {
+  'district_hospital': 'manager',
+  'health_center': 'chw',
+  'clinic': 'member_eligible_woman'
+};
+
+const generateHierarchy = (type, placeName, numberOfPersons,
+  parentFirstLevelId, parentSecondLevelId, directParentPlace) => {
+
+  const place = bracPlaceFactory.generateBracPlace(placeName, type,
+    setParents(parentFirstLevelId, parentSecondLevelId));
+
+  const needUser = pairPlaceTypesNeedsUsers[type];
+  let parentPersonThirdLevelId = null;
+  let parentPersonSecondLevelId = null;
+  let parentPersonFirstLevelId = null;
+
+  if (type === 'district_hospital') {
+    parentPersonFirstLevelId = place._id;
+  }
+  if (type === 'health_center') {
+    parentPersonFirstLevelId = place._id;
+    parentPersonSecondLevelId = parentFirstLevelId;
+    const randomIndex = Faker.faker.datatype.number(numberOfManagersPerDistrictHospital - 1);
+    place.supervisor = managers[randomIndex]._id;
+  }
+  if (type === 'clinic') {
+    parentPersonFirstLevelId = place._id;
+    parentPersonSecondLevelId = parentFirstLevelId;
+    parentPersonThirdLevelId = parentSecondLevelId;
+  }
+
+  let isPrimaryContact = true;
+  for (let i = 0; i < numberOfPersons; i++) {
```

Comment:
This is a veeery big function, please break it appart into smaller chunks. 
For example, extract the contents of this for in a function: 
```javascript
const generatePerson = (type, place, isPrimaryContact) => { .... };
const generateUser = (person) => { ... };
const generateReports = (person) => { ... };
```

So this turns into: 
```javascript
const primaryContact = generatePerson(type, place, true);
if (needUser) {
  generateUser(primaryContact);
}
for (let i = 0; i < numberOfPersons - 1; i++) {
  generatePerson(type, place, false);
}
```